homeassistant/components/tile/__init__.py

```diff
@@ -33,6 +33,32 @@ async def async_setup_entry(hass, entry):
     hass.data[DOMAIN][DATA_COORDINATOR][entry.entry_id] = {}
     hass.data[DOMAIN][DATA_TILE][entry.entry_id] = {}
 
+    # The existence of shared Tiles across multiple accounts requires an entity ID
+    # change:
+    #
+    # Old: tile_{uuid}
+    # New: tile_{username}_{uuid}
+    #
+    # Find any entities with the old format and update them:
+    ent_reg = entity_registry.async_get(hass)
+    for entity in [
+        e
+        for e in ent_reg.entities.values()
+        if e.platform == DOMAIN
+        and not e.unique_id.startswith(f"{DOMAIN}_{entry.data[CONF_USERNAME]}")
+    ]:
+        split = f"{DOMAIN}_"
+        new_unique_id = (split + f"{entry.data[CONF_USERNAME]}_").join(
```

Comment:
I don't think we should use `DOMAIN` as part of `unique_id`. It's not necessary. Sorry I didn't write about this earlier, busy day, not enough coffee 😉